html/modules/custom/reliefweb_api/src/Commands/ReliefWebApiCommands.php

```diff
@@ -300,4 +298,68 @@ public function replace($bundle, $newtag, $oldtag = NULL, array $options = [
     }
   }
 
+  /**
+   * Re-index queued content.
+   *
+   * @command reliefweb-api:reindexqueue
+   *
+   * @usage reliefweb-api:reindexqueue
+   *   Re-index the queue of terms to update.
+   *
+   * @validate-module-enabled reliefweb_api
+   *
+   * @aliases rw-api:reindexqueue
+   */
+  public function reIndexQueue() {
+
+    $reindex_queue = json_decode($this->state->get('reliefweb_api.reindex_queue'));
+
+    if (empty($reindex_queue)) {
+      return;
+    }
+
+    // Use default values from $this->index().
+    $options = [
+      'elasticsearch' => '',
+      'base-index-name' => '',
+      'website' => '',
+      'limit' => 0,
+      'offset' => 0,
+      'filter' => '',
+      'chunk-size' => 500,
+      'tag' => '',
+      'id' => 0,
+      'remove' => FALSE,
+      'replace' => FALSE,
+      'alias' => FALSE,
+      'alias-only' => FALSE,
+      'log' => 'echo',
+      'count-only' => FALSE,
+      'memory-limit' => '512M',
+      'replicas' => NULL,
+      'shards' => NULL,
+    ];
+    foreach (array_keys($reindex_queue) as $bundle) {
+      if (empty($reindex_queue[$bundle])) {
+        continue;
+      }
+      // Launch the re-indexing.
+      try {
+        $options['filter'] = $bundle . ':' . implode(',', $reindex_queue[$bundle]);
+        if ($bundle == 'country' || $bundle == 'source') {
+          $this->index('training', $options);
+          $this->index('job', $options);
+        }
+        $this->index('report', $options);
```

Comment:
There are more things to re-indexed based on the vocabulary (for example source and disaster when the bundle is "country".).


The method below should return all the bundles that need to be re-indexed for the queued vocabulary:

```php
  /**
   * Get the list of bundles to re-index for the given taxonomy vocabulary.
   *
   * @param string $vocabulary
   *   Vocabulary.
   *
   * @return array
   *   List of bundles to re-index.
   */
  protected static function getBundlesToReindex($vocabulary) {
    // Gather a list of all entity reference fields.
    $entity_reference_fields = \Drupal::service('entity_field.manager')
      ->getFieldMapByFieldType('entity_reference');

    // Generate the list of field config IDs for the reference fields.
    $ids = [];
    foreach ($entity_reference_fields as $entity_type_id => $fields) {
      foreach ($fields as $field_name => $info) {
        foreach ($info['bundles'] as $entity_bundle) {
          $ids[] = "$entity_type_id.$entity_bundle.$field_name";
        }
      }
    }

    // Load the configuration of the entity reference fields.
    $field_configs = \Drupal::service('entity_type.manager')
      ->getStorage('field_config')
      ->loadMultiple($ids);

    // Check if the entity reference fields reference the vocabulary in which
    // case retrieve the list of bundles using those fields. Those bundles are
    // the one to re-index.
    $bundles = [];
    foreach ($field_configs as $field_config) {
      $field_name = $field_config->getName();
      $entity_type_id = $field_config->getTargetEntityTypeId();

      if ($field_config->getSetting('target_type') === 'taxonomy_term') {
        $handler_settings = $field_config->getSetting('handler_settings');
        if (isset($handler_settings['target_bundles'][$vocbulary])) {
          $bundles += $entity_reference_fields[$entity_type_id][$field_name]['bundles'];
        }
      }
    }

    // Exclude bundles that are not indexed in the API.
    return array_keys(array_intersect_key(Bundles::$bundles, $bundles));
  }
  ```
  
 Ideally, this should be converted to a non static method with the `\Drupal::service()` calls replaced with proper dependency injection.